showcase/gapic-showcase/src/test/java/com/google/showcase/v1beta1/it/ITOtelMetrics.java

```diff
@@ -0,0 +1,490 @@
+import io.opentelemetry.sdk.metrics.data.PointData;
+import io.opentelemetry.sdk.testing.exporter.InMemoryMetricReader;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+
+/**
+ * Showcase Test to confirm that metrics are being collected and that the correct metrics are being
+ * recorded. Utilizes an in-memory metric reader to collect the data.
+ */
+public class ITOtelMetrics {
+  private static final String ATTEMPT_COUNT = "attempt_count";
+  private static final String OPERATION_COUNT = "operation_count";
+  private static final String ATTEMPT_LATENCY = "attempt_latency";
+  private static final String OPERATION_LATENCY = "operation_latency";
+  private static final String DEFAULT_LANGUAGE = "Java";
+  private InMemoryMetricReader inMemoryMetricReader;
+  private EchoClient grpcClient;
+  private EchoClient httpClient;
+
+  private OpentelemetryMetricsRecorder createOtelMetricsRecorder(
+      InMemoryMetricReader inMemoryMetricReader) {
+    SdkMeterProvider sdkMeterProvider =
+        SdkMeterProvider.builder().registerMetricReader(inMemoryMetricReader).build();
+
+    OpenTelemetry openTelemetry =
+        OpenTelemetrySdk.builder().setMeterProvider(sdkMeterProvider).build();
+
+    // Meter Creation
+    Meter meter =
+        openTelemetry
+            .meterBuilder("ShowcaseOtelMetrics")
+            .setInstrumentationVersion(GaxProperties.getGaxVersion())
+            .build();
+    // OpenTelemetry Metrics Recorder
+    return new OpentelemetryMetricsRecorder(meter);
+  }
+
+  @Before
+  public void setup() throws Exception {
+    inMemoryMetricReader = InMemoryMetricReader.create();
+    OpentelemetryMetricsRecorder otelMetricsRecorder =
+        createOtelMetricsRecorder(inMemoryMetricReader);
+    grpcClient =
+        TestClientInitializer.createGrpcEchoClientOpentelemetry(
+            new MetricsTracerFactory(otelMetricsRecorder));
+    httpClient =
+        TestClientInitializer.createHttpJsonEchoClientOpentelemetry(
+            new MetricsTracerFactory(otelMetricsRecorder));
+  }
+
+  @After
+  public void cleanup() throws InterruptedException {
+    inMemoryMetricReader.shutdown();
+
+    grpcClient.close();
+    httpClient.close();
+
+    grpcClient.awaitTermination(TestClientInitializer.AWAIT_TERMINATION_SECONDS, TimeUnit.SECONDS);
+    httpClient.awaitTermination(TestClientInitializer.AWAIT_TERMINATION_SECONDS, TimeUnit.SECONDS);
+  }
+
+  private void verifyMetricData(
+      List<MetricData> metricDataList, int operationCount, int attemptCount) {
+    for (MetricData metricData : metricDataList) {
+      Data<?> data = metricData.getData();
+      List<PointData> points = new ArrayList<>(data.getPoints());
+      Truth.assertThat(points.size()).isEqualTo(1);
+      PointData pointData = points.get(0);
```

Comment:
When testing this, all the metrics have a list of PointData entries and the size of the list is one. My assumption with this is that normally the OtelSdk is created with a PeriodicMetricReader as opposed to the InMemoryMetricsReader used in this test. The PeriodicMetricReader would be reading the data as set intervals and each interval would be a new PointData object (i.e. If the interval is 10s and data is read over 60s, there would be a list of 6 PointData entries).

I think the `collectAllMetrics()` aggregates the PointData values together for each metrics and this is why there is only one entry in the list. I couldn't find anything in the Otel docs to confirm this though.